crates/matrix-sdk-base/src/rooms/normal.rs

```diff
@@ -596,24 +596,39 @@ impl Room {
                 let name = name.trim();
                 return Ok(DisplayName::Named(name.to_owned()));
             }
+
             if let Some(alias) = inner.canonical_alias() {
                 let alias = alias.alias().trim();
                 return Ok(DisplayName::Aliased(alias.to_owned()));
             }
+
             inner.summary.clone()
         };
 
         let own_user_id = self.own_user_id().as_str();
 
-        let members: Vec<RoomMember> = if summary.heroes.is_empty() {
-            self.members(RoomMemberships::ACTIVE)
-                .await?
-                .into_iter()
-                .filter(|u| u.user_id() != own_user_id)
-                .take(5)
-                .collect()
+        let (heroes, guessed_num_members): (Vec<RoomMember>, _) = if summary.heroes.is_empty() {
+            let mut members = self.members(RoomMemberships::ACTIVE).await?;
+
+            // Make the ordering deterministic.
+            members.sort_unstable_by(|lhs, rhs| lhs.name().cmp(rhs.name()));
+
+            // We can make a good prediction of the total number of members here. This might
+            // be incorrect if the database info is outdated.
+            let guessed_num_members = Some(members.len());
+
+            // The server returns at most 5 heroes; behave similarly here.
+            (
+                members.into_iter().take(5).filter(|u| u.user_id() != own_user_id).collect(),
+                guessed_num_members,
+            )
         } else {
-            let members: Vec<_> = stream::iter(summary.heroes.iter())
+            let mut heroes = summary.heroes;
+
+            // Make the ordering deterministic.
+            heroes.sort();
```

Comment:
We are using `sort_unstable` above, and here we are using `sort`, is that fine?

Reply:
Yeah, this code path could be used without the other code path being called, so we need to sort; at most there are 5 items in that list, so any sorting works. Or are you requesting to use `sort_unstable` instead of `sort` for consistency? (Wouldn't make a big difference, but happy to make the change)

Reply:
The latter. I think it's better for the sake of consistency.